components/director/internal/domain/runtime/resolver.go

```diff
@@ -251,11 +273,17 @@ func (r *Resolver) UpdateRuntime(ctx context.Context, id string, in graphql.Runt
 
 // DeleteRuntime missing godoc
 func (r *Resolver) DeleteRuntime(ctx context.Context, id string) (*graphql.Runtime, error) {
+	var selfRegLabelVal string
 	tx, err := r.transact.Begin()
 	if err != nil {
 		return nil, err
 	}
-	defer r.transact.RollbackUnlessCommitted(ctx, tx)
+	defer func() {
+		didRollback := r.transact.RollbackUnlessCommitted(ctx, tx)
+		if !didRollback { // if we did rollback we should not try to execute the cleanup
```

Comment:
The value of the label is returned by the external service. We cannot calculate that. Also, if we try to get the label value in `r.cleanupAndLogOnError(ctx, selfRegLabelVal)` we would not find it because once the runtime is deleted the cascade will delete the label as well.